CLI getting started updates for RC4

mairaw opened the pull request:
Fixes #1506 

Made updates to use the new dotnet new command and updated templates, refer to code snippets in the samples folder, among other things (for example, introduced the concept of using the arguments just in the second example and removed the different namespace name from the last section since I don't think it's common practice to have different namespaces names within the same project.

To be merged only when RC4 is out.

/cc @blackdwarf

mairaw committed:
CLI getting started updates for RC4

blackdwarf commented:
@mairaw hold off for a spell. We are talking about the `dotnet new` errr...new experience and whether it will change.

mairaw committed:
fixed sample names

mairaw commented:
I'm not merging this until RC4 is out the door. keep me posted @blackdwarf

spboyer commented:
@mairaw are you also updating the step by step as well? There are some needed changes in that too.

mairaw commented:
which one @spboyer? this one is included in this PR: https://docs.microsoft.com/en-us/dotnet/articles/core/preview3/tutorials/using-with-xplat-cli-msbuild

spboyer commented:
yes, that one. Missed it in the files included. 👍

mairaw commented:
@blackdwarf looks good? ok to merge?

mairaw commented on docs/core/preview3/tutorials/using-with-xplat-cli-msbuild.md:
````diff
@@ -1,190 +1,107 @@
     ```
 
-    Alternatively, you can also execute [`dotnet build`](../tools/dotnet-build.md) to compile the code without running the build console applications. This results in a `bin/Debug/netcoreapp1.0/Hello.dll` compiled application that can be run with `dotnet bin\Debug\netcoreapp1.0\Hello.dll` on Windows, and `dotnet bin/Debug/netcoreapp1.0/Hello.dll` on other systems. You may specify an additional pa ...
+    Alternatively, you can also execute [`dotnet build`](../tools/dotnet-build.md) to compile the code without running the build console applications. This results in a *bin/Debug/netcoreapp1.0/Hello.dll* compiled application that can be run with `dotnet bin\Debug\netcoreapp1.0\Hello.dll` on Windows, and `dotnet bin/Debug/netcoreapp1.0/Hello.dll` on other systems. You may specify an additional pa ...
````
this paragraph seems to be inaccurate since the running is the same for both Windows and non-Windows and I didn't know how to change. @blackdwarf can you help?

spboyer replied:
@mairaw the command is the same, however the the `/` vs `\` is the difference. I'm not sure how relevant that paragraph is, I would remove it.

mairaw replied:
ah completely missed the forward and backslash @spboyer. Thanks!

mairaw committed:
clarifying the difference

mairaw committed:
a few more changes

stevehoag commented:
LGTM :shipit:

blackdwarf commented on samples/core/console-apps/FibonacciBetterMsBuild/README.md:
```diff
@@ -0,0 +1,25 @@
+To build and run the sample, type the following three commands:
+
+`dotnet restore`
+`dotnet build`
```
You don't need this one. `run` will `build` for you.

blackdwarf commented on samples/core/console-apps/FibonacciBetterMsBuild/README.md:
```diff
@@ -0,0 +1,25 @@
+`dotnet build`
+`dotnet run`
+
+`dotnet restore` installs all the dependencies for this sample into the current directory.
```
Not true, better would be "restores the dependencies for this sample". They don't really go into the current directory and maybe at this point we don't care where they go?

blackdwarf commented on samples/core/console-apps/FibonacciBetterMsBuild/README.md:
```diff
@@ -0,0 +1,25 @@
+`dotnet run`
+
+`dotnet restore` installs all the dependencies for this sample into the current directory.
+`dotnet build` creates the output assembly (or assemblies).
```
Same as the above comment. Don't need this yet.

blackdwarf commented on samples/core/console-apps/fibonacci-msbuild/README.md:
```diff
@@ -0,0 +1,24 @@
+To build and run the sample, type the following three commands:
+
+`dotnet restore`
+`dotnet build`
```
As above.

thesmithee reviewed (changes requested):
Added some changes. Now, we need and maybe we should somehow add `dotnet publish` and `dotnet path/to/assembly.dll` here to make sure people see it.

mairaw committed:
fixed readme files

mairaw commented:
That was kind of the template we were using for all the samples readme files @blackdwarf. I've changed all the applicable ones to remove dotnet build and changed the description for dotnet run and dotnet restore, including the C# samples from @BillWagner. 

So for each one of them, do you also want to add the alternative `dotnet path/to/dll` and `dotnet publish`? That starts to complicate things a bit, no? Path to DLL will depend on the version of the runtime you have installed, so I cannot give users the exact command. Do I need to add the dotnet build step back so I show the two paths: restore/run and restore/build/dotnet dll?

And for dotnet publish, which one would you like to add to the readme? self-contained ones? 

Let me know what you think @blackdwarf.

blackdwarf commented:
@mairaw I see. That is weird, `dotnet build` is not needed in that case as `run` will build your code. Sorry about that, I seem to have misunderstood in my haste what the change is. I was only referring to the getting started document. But since we are already talking about it...do you think that removing `dotnet build` in this case makes sense especially w.r.t the #1509 issue?

mairaw commented:
@blackdwarf the change was only for getting started initially. For that work, I did include new samples to refer them from the getting started topic so they can be tested as well vs. just pasted in the topic. with that, I've just used the template readme file we were using for all other sample which showed dotnet restore, dotnet build, dotnet run. I agree with you that the dotnet build is not needed so I just fixed those across all readmes. 

@BillWagner what do you think?

mairaw committed:
Merge branch 'master' into rc4

mairaw commented:
It would be good to get this out with the new syntax for dotnet new. I've added info into the topic about development time, etc. @blackdwarf any changes you still would like to see?
